sympy/sets/tests/test_fancysets.py

```diff
@@ -236,24 +236,24 @@ def test_Range_set():
     assert Range(2, -oo, -2)[:2:2] == Range(2, -2, -4)
     raises(ValueError, lambda: Range(-oo, 4, 2)[:2:2])
     assert Range(-oo, 4, 2)[::-2] == Range(2, -oo, -4)
-    raises(ValueError, lambda: Range(-oo, 4, 2)[::2])
+    assert Range(-oo, 4, 2)[::2] == Range(-oo, 4, 4)
```

Comment:
I don't think this is well defined. If I wanted to use Range slicing I can imagine that this behaviour could be an annoying corner case that I might need to work around rather than something useful.
```
In [5]: Range(-oo, 4, 2)                                                                                                          
Out[5]: {…, 0, 2}

In [6]: Range(-oo, 4, 2)[::2]                                                                                                     
Out[6]: {…, -4, 0}
```

Reply:
@smichr Please provide your comments on this, so that I can make the changes accordingly.
I think `Range(-oo, 4, 4)` is fine. Please clarify more on this, like how can this behaviour cause troubles.

Reply:
It just isn't well defined so why define it?
```julia
In [35]: Range(0, 3)                                                                                                              
Out[35]: {0, 1, 2}

In [36]: Range(-oo, 3)                                                                                                            
Out[36]: {…, 1, 2}

In [37]: Range(0, 3)[::2]                                                                                                         
Out[37]: {0, 2}

In [38]: Range(-oo, 3)[::2]                                                                                                       
Out[38]: {…, -1, 1}
```
Maybe one day we will have a symbolic Slice and we will be able to represent things like:
```julia
In [39]: Range(m, n)[::2]            # Raises under this PR                                                                                             
---------------------------------------------------------------------------
NotImplementedError
```
If that happens the results here will be inconsistent with substitutions like `Range(n, 3)[::2].subs(n, oo) == Range(n, 3).subs(n, oo)[::2]`. Right now in this PR you have disabled slicing in that case - if we had a Slice object then we would just return that instead. I think slicing with infinite start should only be defined if the step is +-1.

Do you have any use case motivating why this should be defined as it is? Is it actually used anywhere in SymPy?